From 4046a9184f703d0378ec3ce3f8b98fdccdc9ba2a Mon Sep 17 00:00:00 2001
From: Guilhem IMBERTON <guilhem.imberton@intel.com>
Date: Thu, 4 Oct 2012 14:26:57 +0200
Subject: [PORT FROM MAIN] Use consistent search depth.

BZ: 61996

AndroidProducts.mk allows for a maxdepth of 6 in device and vendor while
vendorsetup.sh and BoardConfig.mk only allow for smaller depth.

Make search depth consistent for these files.

Change-Id: Ifec567b493d29fe1c1fc094849ad98bb73395e82
Orig-Change-Id: Ib2b37b7eefd21c8edf7ebb847074f42f3f15db99
Signed-off-by: Guilhem IMBERTON <guilhem.imberton@intel.com>
Signed-off-by: Laurent FERT <laurent.fert@intel.com>
Reviewed-on: http://android.intel.com:8080/79987
---
 core/config.mk |    5 ++---
 envsetup.sh    |    1 -
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/core/config.mk b/core/config.mk
index 58d3900..7f60aea 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -143,9 +143,8 @@ include $(BUILD_SYSTEM)/envsetup.mk
 board_config_mk := \
 	$(strip $(wildcard \
 		$(SRC_TARGET_DIR)/board/$(TARGET_DEVICE)/BoardConfig.mk \
-		device/*/$(TARGET_DEVICE)/BoardConfig.mk \
-		vendor/*/$(TARGET_DEVICE)/BoardConfig.mk \
-		vendor/*/*/board/$(TARGET_DEVICE)/BoardConfig.mk \
+		$(shell test -d device && find device -maxdepth 6 -path '*/$(TARGET_DEVICE)/BoardConfig.mk') \
+		$(shell test -d vendor && find vendor -maxdepth 6 -path '*/$(TARGET_DEVICE)/BoardConfig.mk') \
 	))
 ifeq ($(board_config_mk),)
   $(error No config file found for TARGET_DEVICE $(TARGET_DEVICE))
diff --git a/envsetup.sh b/envsetup.sh
index fdb9f79..5fe6df2 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -1265,7 +1265,6 @@ if [ "x$SHELL" != "x/bin/bash" ]; then
 fi
 
 # Execute the contents of any vendorsetup.sh files we can find.
-#for f in `/bin/ls vendor/*/vendorsetup.sh vendor/*/*/vendorsetup.sh device/*/*/vendorsetup.sh 2> /dev/null`
 for f in `test -d device && find device -maxdepth 6 -name 'vendorsetup.sh' 2> /dev/null` \
          `test -d vendor && find vendor -maxdepth 6 -name 'vendorsetup.sh' 2> /dev/null`
 do
-- 
1.7.4.1

